front/components/data_source/DatasourceTablesTabView.tsx

```diff
@@ -24,98 +70,151 @@ export function DatasourceTablesTabView({
 }) {
   const [showTableUploadModal, setShowTableUploadModal] = useState(false);
   const [tableToLoad, setTableToLoad] = useState<string | null>(null);
+  const [showTableDeleteDialog, setShowTableDeleteDialog] = useState(false);
+  const [tableSearch, setTableSearch] = useState<string>("");
 
-  const { tables } = useTables({
+  const sendNotification = useContext(SendNotificationsContext);
+
+  const { tables, mutateTables } = useTables({
     workspaceId: owner.sId,
     dataSourceName: dataSource.name,
   });
 
+  const handleDelete = async (tableId: string) => {
+    try {
+      const res = await fetch(
+        `/api/w/${owner.sId}/data_sources/${dataSource.name}/tables/${tableId}`,
+        {
+          method: "DELETE",
+        }
+      );
+
+      if (!res.ok) {
+        throw new Error("Failed to delete table");
+      }
+
+      sendNotification({
+        type: "success",
+        title: "Table successfully deleted",
+        description: `The table was successfully deleted`,
+      });
+
+      await mutateTables();
+    } catch (error) {
+      sendNotification({
+        type: "error",
+        title: "Error deleting table",
+        description: "An error occurred while deleting the table.",
+      });
+    }
+  };
+
+  const columns = [
+    {
+      accessorKey: "name",
+      header: "Name",
+      cell: (info: Info) => (
+        <DataTable.Cell icon={ServerIcon}>
+          {info.row.original.name}
+        </DataTable.Cell>
+      ),
+    },
+    {
+      header: "Last Edited",
+      accessorKey: "lastEdited",
+      cell: (info: Info) => (
+        <DataTable.Cell>
+          {timeAgoFrom(info.row.original.timestamp)} ago
+        </DataTable.Cell>
+      ),
+    },
+    {
+      id: "actions",
+      cell: (info: Info) => (
+        <EditOrDeleteDropdown
+          onEdit={() => {
+            setTableToLoad(info.row.original.tableId);
+            setShowTableUploadModal(true);
+          }}
+          onDelete={() => {
+            setTableToLoad(info.row.original.tableId);
+            setShowTableDeleteDialog(true);
+          }}
+        />
+      ),
```

Comment:
There's actually a specific prop in DataTable ( `onMoreClick` ) for this action thing, you don't need to define column. However it's just an action handler, don't know how it can interact with a dropdown menu ? But we should fix it so that it's usable to open a menu, or completely drop it if we think it's not usable

Reply:
Yeah a bit annoying to use the onMoreClick here..